### PR TITLE
Preparation for v2024.2 release (Engine)

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This PR is a preparation for v2024.2 release.

## Changes
*  Changes
* lib/Zonemaster/Engine.pm
* Makefile.PL

## Note

CI will fail until https://github.com/zonemaster/zonemaster-ldns/pull/214 is merged.

## How to test this PR

Review.

### CODE REVIEW

**Comment by matsduf** [comment]
CI fails until https://github.com/zonemaster/zonemaster-ldns/pull/214 is merged.

**Review by marc-vanderwal — DISMISSED** [review]

File: Changes [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+ - Fixes CDS and CDNSKEY RRsets comparison in test case DNSSEC15 (#1383)
+ - Removes test case Nameserver14 never implemented (#1390)
+ - Updates test case Delegation05 to include QNAME and QTYPE in diagnostics for
+   failed queries (#1392)
+ - Removes obsolete document (#1359)
+ - Resolves IP addresses directly from Engine for out-of-bailiwick names in fake
+   delegations (#1389)
+ - Fixes spelling errors (external controbution from @emollier) (#1378)
+ - Fix barewords (external controbution from @emollier) (#1380)
+ - Fixes misspellings (external controbution from @jsoref) (#1366)
```
Comment (mattias-p) [inline Changes:38]: Misspellings in entries about fixing misspellings.
```suggestion
 - Fixes spelling errors (external contribution from @emollier) (#1378)
 - Fix barewords (external contribution from @emollier) (#1380)
 - Fixes misspellings (external contribution from @jsoref) (#1366)
```

File: Changes [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+
+ [Breaking change]
+ - Refactors ASNLookup code and documentation (#1257)
+
+ [Features]
+ - Changes default settings of queries (#1397)
+ - Updates DNSSEC10 implementation (#1396)
+ - Updates global cache and makes the feature being supported and not
+   experimental (#1394)
+ - Update Connectivity04 implementation (#1393)
```
Comment (mattias-p) [inline Changes:18]: This should probably be moved to the Fixes category. The PR is marked with V-Patch and at least to me it feels more like a patch than a minor.

**Review by mattias-p — APPROVED** [review]

File: Changes (hunk 2 above)
Comment (matsduf) [reply Changes]: Fixed.

**Comment by matsduf** [comment]
@mattias-p, @marc-vanderwal, please re-approve after editorial changes.

**Comment by matsduf** [comment]
@mattias-p, @marc-vanderwal and @tgreenx: please (re-)approve after editorial changes.

**Review by marc-vanderwal — APPROVED** [review]

**Review by mattias-p — APPROVED** [review]

**Pull request merged**